src/main/java/at/hannibal2/skyhanni/data/title/TitleManager.kt

```diff
@@ -0,0 +1,351 @@
+package at.hannibal2.skyhanni.data.title
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.api.event.HandleEvent
+import at.hannibal2.skyhanni.config.commands.CommandCategory
+import at.hannibal2.skyhanni.config.commands.CommandRegistrationEvent
+import at.hannibal2.skyhanni.config.commands.brigadier.BrigadierArguments
+import at.hannibal2.skyhanni.config.core.config.Position
+import at.hannibal2.skyhanni.data.title.CountdownTitleContext.Companion.fromTitleData
+import at.hannibal2.skyhanni.events.DebugDataCollectEvent
+import at.hannibal2.skyhanni.events.GuiRenderEvent
+import at.hannibal2.skyhanni.events.minecraft.SkyHanniTickEvent
+import at.hannibal2.skyhanni.events.minecraft.WorldChangeEvent
+import at.hannibal2.skyhanni.skyhannimodule.SkyHanniModule
+import at.hannibal2.skyhanni.test.command.ErrorManager
+import at.hannibal2.skyhanni.utils.ChatUtils
+import at.hannibal2.skyhanni.utils.InventoryUtils
+import at.hannibal2.skyhanni.utils.TimeUtils
+import at.hannibal2.skyhanni.utils.collection.CollectionUtils
+import at.hannibal2.skyhanni.utils.collection.CollectionUtils.enumMapOf
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.milliseconds
+import kotlin.time.Duration.Companion.seconds
+
+@SkyHanniModule
+object TitleManager {
+
+    private val titleLocationQueues: MutableMap<TitleLocation, CollectionUtils.OrderedQueue<TitleContext>> = enumMapOf()
+    private val currentTitles: MutableMap<TitleLocation, TitleContext?> = enumMapOf()
+    val guiConfig get() = SkyHanniMod.feature.gui
+    val existingIntentions = guiConfig.titleIntentionPositions.values.map { it.keys }.flatten().toSet()
+    val intentionMapper: MutableMap<String, TitleIntention> = mutableMapOf()
+
+    inline fun<reified E : Enum<E>> registerIntentions(
```

Comment:
what is with this 30 lines unused function here?

Reply:
Classes need to register their intentions with an enum to use them - no implementations of intentions in this PR, so unused